jt-classbreaks/src/main/java/it/geosolutions/jaiext/classbreaks/QuantileBreaksHistogramOpImage.java

```diff
@@ -102,8 +99,24 @@ protected void postCalculate(Classification c, int band) {
                 classIdx++;
                 set.add(e.getMin());
             }
+
         }
         set.add(buckets.get(buckets.size() - 1).getMax());
         hc.setBreaks(band, set.toArray(new Double[set.size()]));
+        if (this.percentages.booleanValue()) {
+            int nBreaks = set.size();
+            int actualClassNum = numClasses >= nBreaks ? nBreaks - 1 : numClasses;
+            hc.setPercentages(getPercentages(new ArrayList<>(set), buckets, nvalues, actualClassNum));
+        }
+    }
+
+    private double[] getPercentages(List<Double> tBreaks, List<Bucket> buckets, int nvalues, int numClasses) {
+        Map<Double, Integer> values = new HashMap<>();
+        for (Bucket b : buckets) {
+            values.put(b.getMin(), b.getCount());
+        }
+        ClassPercentagesManager percentagesManager = new ClassPercentagesManager();
+        return percentagesManager.getPercentages(values, tBreaks, nvalues, numClasses);
```

Comment:
Both histogram based natural breaks and quantiles percentage calculation require counting portions of buckets in a given set of ranges, that do not necessarily match the buckets limits. So why are the two using two different methods to compute the results?

Reply:
fixed